examples/instance/terraform/plugin.go

```diff
@@ -370,8 +370,25 @@ func (p *plugin) Provision(spec instance.Spec) (*instance.ID, error) {
 			}
 		}
 	case "softlayer_virtual_guest":
-		log.Debugln("softlayer_virtual_guest detected, adding hostname to properties: hostname=", name)
-		properties.Value["hostname"] = name
+		// Use the given hostname value as a prefix if it is a non-empty string
```

Comment:
I think we can move this *entire* block to be outside the `switch` so that it can work for *all* resource types.   This is useful in general and the magic property is removed anyway.

Also, to indicate this `hostname_prefix` is a magic property, how about adopting a convention like `@hostname_prefix`?  This would make it easy for user to know which properties are essentially processing directives and not actual data.